### PR TITLE
Add support for deletion or preservation of resource group tags.

### PR DESCRIPTION
**Pull request by StylusEaterChef**
Thanks @devopsdina!

Signed-off-by: Adam M Dutko <adutko@chef.io>

### Description

Add support as requested while preserving current behavior.

### Issues Resolved

[List any existing issues this PR resolves, or any Discourse or
StackOverflow discussions that are relevant]

#109 

### Check List

- [ ] New functionality includes tests
- [X] All tests pass
- [X] PR title is a worthy inclusion in the CHANGELOG

### CODE REVIEW

**Comment by stuartpreston** [comment]
Just chiming in here (as I left Chef and this project earlier this year). Thanks for this contribution, it looks fine to me though probably could be refactored a little to reduce duplicate code between true and false values -- but then, so could the whole driver..!

I would request/suggest that the Readme is updated with the new key and default value, then hopefully someone on the relevant team at Chef can pick it up for merge. (cc: @tas50)

**Comment by StylusEaterChef** [comment]
> Just chiming in here (as I left Chef and this project earlier this year). Thanks for this contribution, it looks fine to me though probably could be refactored a little to reduce duplicate code between true and false values -- but then, so could the whole driver..!

I'm doing more refactoring as I've found some state related bugs. I'm hoping to clean things up as I go because I'd hate to do some big refactor and break stuff.

> I would request/suggest that the Readme is updated with the new key and default value, then hopefully someone on the relevant team at Chef can pick it up for merge. (cc: @tas50)

You got it. That's next up on the list.

**Comment by StylusEaterChef** [comment]
@tas50 and @stuartpreston README updated. I've cleaned up the code through another issue.

File: .gitignore [hunk 1]
```diff
@@ -4,3 +4,6 @@ Gemfile.lock
 /docs/examples
 coverage
 *.log
+kitchen.local.yml
```
Comment (tas50) [inline .gitignore:7]: dont both fixing it, but this could be .*kitchen.local.yml

File: lib/kitchen/driver/azurerm.rb [hunk 2]
```diff
@@ -706,6 +736,13 @@ def prepared_custom_data
       # Wrapper methods for the Azure API calls to retry the calls when getting timeouts.
       #
 
+      def get_resource_group
```
Comment (tas50) [inline lib/kitchen/driver/azurerm.rb:739]: can you toss some yard on this method so we know what it returns

**Pull request merged**